packages/flutter_tools/gradle/aar_init_script.gradle

```diff
@@ -0,0 +1,117 @@
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.maven.MavenDeployer
+import org.gradle.api.plugins.MavenPlugin
+import org.gradle.api.tasks.Upload
+
+void configureProject(Project project, File outputDir) {
+    if (!project.hasProperty("android")) {
+        throw new GradleException("Android property not found.")
+    }
+    if (!project.android.hasProperty("libraryVariants")) {
+        throw new GradleException("Can't generate AAR on a non Android library project.");
+    }
+
+    project.apply plugin: "maven"
+
+    project.android.libraryVariants.all { variant ->
+        addAarTask(project, variant)
+    }
+    // Snapshot versions include the timestamp in the artifact name.
+    project.version = project.version.replace("-SNAPSHOT", "")
+
+    project.uploadArchives {
+        repositories {
+            mavenDeployer {
+                repository(url: "file://${outputDir}/outputs/repo")
+            }
+        }
+    }
+    Boolean usesFlutterPlugin = project.plugins.find { it.class.name == "FlutterPlugin" } != null
+    if (!usesFlutterPlugin) {
+        // Plugins don't include their dependencies under the assumption that the parent project adds them.
+        if (project.properties['android.useAndroidX']) {
+            project.dependencies {
+                compileOnly "androidx.annotation:annotation:+"
+            }
+        } else {
+            project.dependencies {
+                compileOnly "com.android.support:support-annotations:+"
+            }
+        }
+        project.dependencies {
+            // The Flutter plugin already adds `flutter.jar`.
+            compileOnly project.files("${getFlutterRoot(project)}/bin/cache/artifacts/engine/android-arm-release/flutter.jar")
```

Comment:
Not for this PR, but it would probably be nice to have a version of flutter.jar that is interface only.  The engine already produces an intermediary for that and then adds in libflutter.so.  We could probably make a step where we also just produce a flutter.jar that just has the Java classes and no libflutter.so, so we don't end up in this strange scenario where we're picking a specific build variant that isn't necessarily what the final build will use.

Reply:
I like that idea. 👍